src/miniscript/satisfy.rs

```diff
@@ -947,6 +947,44 @@ impl Satisfaction {
                     }
                 }
             }
+            Terminal::MultiA(k, ref keys) => {
+                // Collect all available signatures
+                let mut sig_count = 0;
+                let mut sigs = vec![vec![vec![]]; keys.len()];
+                for (i, pk) in keys.iter().enumerate() {
+                    match Witness::signature(stfr, pk) {
+                        Witness::Stack(sig) => {
+                            sigs[i] = sig;
+                            sig_count += 1;
+                            // This a privacy issue, we are only selecting the first available
+                            // sigs. Incase pk at pos 1 is not selected, we know we did not have access to it
+                            // bitcoin core also implements the same logic for MULTISIG, so I am not bothering
+                            // permuting the sigs for now
```

Comment:
In c05eb7c70d087269cc52c156b4ae4eadd2657133:

Ok to leave this comment here, but TBH I don't think there's any way we could eliminate this privacy leak. If the first signature is missing then the resulting witnesses will have a different distribution than if it's not ... which will become apparent over time.

Although, I guess, in practice users are unlikely to reuse keys in visible ways, so an adversary will have only a one-shot view of the signature distribution.